### PR TITLE
Simplify cudacompat layer to use a 1-dimensional grid

### PR DESCRIPTION
**Pull request by fwyzard**
Remove the possibility of changing the grid size used by the `cms::cudacompat` layer, and make it a constant equal to `{1, 1, 1}`.

This avoids a thread-related problem caused by TBB using worker threads where the grid size had not been initialised.

The kernel for pixel clustering need to be rewritten to support a one-dimensional grid to run on the CPU.
Currently they are only used on the GPU in the Patatrack workflows, but they are exercised on the CPU by the `gpuClustering_t` tests; those tests have been commented out until the kernels can be updated.

### CODE REVIEW

**Comment by fwyzard** [comment]
:construction: Validation running at fu-c2a02-35-01:/data/user/fwyzard/patatrack/validation/run_586.MmsQO5msyA ...

**Comment by fwyzard** [comment]
Should fix #564

**Comment by VinInn** [comment]
if compiles and run the code is ok with me.
CPU workflows may even go faster w/o TLS

File: RecoPixelVertexing/PixelVertexFinding/plugins/gpuVertexFinderImpl.h [hunk 1]
```diff
@@ -157,10 +156,9 @@ namespace gpuVertexFinder {
     // std::cout << "found " << (*ws_d).nvIntermediate << " vertices " << std::endl;
     fitVertices(soa, ws_d.get(), 50.);
     // one block per vertex!
-    blockIdx.x = 0;
-    gridDim.x = 1;
+    assert(blockIdx.x == 0);
```
Comment (VinInn) [inline RecoPixelVertexing/PixelVertexFinding/plugins/gpuVertexFinderImpl.h:159]: we could remove the asserts as this seems the only place (and now they are const!)

**Review by VinInn — APPROVED** [review]

File: RecoPixelVertexing/PixelVertexFinding/plugins/gpuVertexFinderImpl.h [hunk 2]
```diff
@@ -157,10 +156,9 @@ namespace gpuVertexFinder {
...
+    assert(gridDim.x == 1);
```
Comment (fwyzard) [inline RecoPixelVertexing/PixelVertexFinding/plugins/gpuVertexFinderImpl.h:160]:
```suggestion
```

File: RecoPixelVertexing/PixelVertexFinding/test/VertexFinder_t.h [hunk 3]
```diff
@@ -266,10 +266,9 @@ int main() {
       cms::cuda::launch(gpuVertexFinder::splitVerticesKernel, {1024, 64}, onGPU_d.get(), ws_d.get(), 9.f);
       cudaCheck(cudaMemcpy(&nv, LOC_WS(nvIntermediate), sizeof(uint32_t), cudaMemcpyDeviceToHost));
 #else
-      gridDim.x = 1;
+      assert(gridDim.x == 1);
       assert(blockIdx.x == 0);
```
Comment (fwyzard) [inline RecoPixelVertexing/PixelVertexFinding/test/VertexFinder_t.h:270]:
```suggestion
```

**Review by fwyzard — COMMENTED** [review]
remove unnecessary asserts.

File: HeterogeneousCore/CUDAUtilities/interface/cudaCompat.h [hunk 4]
```diff
@@ -11,21 +11,26 @@
 #include <cstdint>
 #include <cstring>
 
+// include the CUDA runtime header to define some of the attivutes, types and sybols also on the CPU
```
Comment (makortel) [inline HeterogeneousCore/CUDAUtilities/interface/cudaCompat.h:14]: Typo?
```suggestion
// include the CUDA runtime header to define some of the attributes, types and sybols also on the CPU
```

**Comment by fwyzard** [comment]
All crashes in the CPU workflows are indeed fixed.

No changes to the physics performance or throughput, as expected.

**Pull request merged**